internal/controller/apiextensions/composite/ready.go

```diff
@@ -226,8 +224,8 @@ func (c ReadinessCheck) IsReady(p *fieldpath.Paved, o ConditionedObject) (bool,
 		return val == *c.MatchInteger, nil
 	case ReadinessCheckTypeMatchCondition:
 		// we should have checked this outside of this function
```

Comment:
This comment too can actually be removed, it feels like this was a default case that got lost during a refactoring